File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
  
  let obj = {}
  for (let i=0; i<players.length; i++) {
    obj[players[i]] = i
  }

  let idx = 0;
  for (let calling of callings) {
    idx = obj[calling]
    obj[calling] -= 1; // 세미콜론 필수
    obj[players[idx-1]] += 1; // 세미콜론 필수
    [players[idx],players[idx-1]] = [players[idx-1],players[idx]];
  };

  return players;
}


console.log(solution(["mumu", "soe", "poe", "kai", "mine"],["kai", "kai", "mine", "mine"]))

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
# 조금 더 다듬어보자
def solution(players, callings):

    dict = {name: idx for idx,name in enumerate(players)}

    for calling in callings:
        rank = dict[calling]
        dict[calling] = rank-1

        # calling앞에 있는 사람을 players배열에서 찾아서 등수1개 올리기
        dict[players[rank-1]] = rank
        players[rank-1], players[rank] = players[rank], players[rank-1]
    
    return players

print(solution(["mumu", "soe", "poe", "kai", "mine"],["kai", "kai", "mine", "mine"]))